python-package/lightgbm/compat.py

```diff
@@ -64,23 +64,38 @@ class DataFrame(object):
     from sklearn.base import RegressorMixin, ClassifierMixin
     from sklearn.preprocessing import LabelEncoder
     from sklearn.utils import deprecated
+    from sklearn.utils.multiclass import check_classification_targets
+    from sklearn.utils.validation import check_X_y, check_array, check_consistent_length
     try:
         from sklearn.model_selection import StratifiedKFold, GroupKFold
+        from sklearn.exceptions import NotFittedError
     except ImportError:
         from sklearn.cross_validation import StratifiedKFold, GroupKFold
+        from sklearn.utils.validation import NotFittedError
     SKLEARN_INSTALLED = True
-    LGBMModelBase = BaseEstimator
-    LGBMRegressorBase = RegressorMixin
-    LGBMClassifierBase = ClassifierMixin
-    LGBMLabelEncoder = LabelEncoder
+    _LGBMModelBase = BaseEstimator
+    _LGBMRegressorBase = RegressorMixin
+    _LGBMClassifierBase = ClassifierMixin
+    _LGBMLabelEncoder = LabelEncoder
     LGBMDeprecated = deprecated
-    LGBMStratifiedKFold = StratifiedKFold
-    LGBMGroupKFold = GroupKFold
+    LGBMNotFittedError = NotFittedError
+    _LGBMStratifiedKFold = StratifiedKFold
+    _LGBMGroupKFold = GroupKFold
+    _LGBMCheckXY = check_X_y
+    _LGBMCheckArray = check_array
+    _LGBMCheckConsistentLength = check_consistent_length
+    _LGBMCheckClassificationTargets = check_classification_targets
 except ImportError:
-    SKLEARN_INSTALLED = False
-    LGBMModelBase = object
-    LGBMClassifierBase = object
-    LGBMRegressorBase = object
-    LGBMLabelEncoder = None
-    LGBMStratifiedKFold = None
-    LGBMGroupKFold = None
+    _SKLEARN_INSTALLED = False
+    _LGBMModelBase = object
+    _LGBMClassifierBase = object
+    _LGBMRegressorBase = object
+    _LGBMLabelEncoder = None
+    LGBMDeprecated = None
```

Comment:
This will raise an error if sklearn not installed, check https://github.com/Microsoft/LightGBM/issues/221. You can uninstall scikit-learn and take a try.

Reply:
I thought that it's just forgotten object. Sorry.
btw, in two words why these dummy variables are needed?

Reply:
Maybe some of these I don't need to specify too?
```
LGBMNotFittedError = ValueError 
_LGBMCheckXY = None
_LGBMCheckArray = None
_LGBMCheckConsistentLength = None
_LGBMCheckClassificationTargets = None 
```

Reply:
I uninstalled skikit-learn, removed the line
`LGBMDeprecated = None`
and
`import lightgbm as lgb`
didn't raise an error. Does it indicate that everything is OK?